datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/system/cdc/debezium/ConfigureDebeziumConnectorStep.java

```diff
@@ -0,0 +1,278 @@
+    this.kafkaProperties = kafkaProperties;
+  }
+
+  @Override
+  public String id() {
+    return "ConfigureDebeziumConnectorStep";
+  }
+
+  @Override
+  public Function<UpgradeContext, UpgradeStepResult> executable() {
+    return (context) -> {
+      log.info("Configuring Debezium connector for CDC processing...");
+
+      try {
+        String connectUrl = debeziumConfig.getUrl();
+        if (connectUrl == null || connectUrl.isEmpty()) {
+          log.error(
+              "Kafka Connect URL not configured in CDC configuration - cannot configure connector");
+          return new DefaultUpgradeStepResult(this.id(), DataHubUpgradeState.FAILED);
+        }
+
+        Map<String, Object> connectorConfig = buildConnectorConfiguration();
+        if (connectorConfig.isEmpty()) {
+          log.error("Failed to build connector configuration");
+          return new DefaultUpgradeStepResult(this.id(), DataHubUpgradeState.FAILED);
+        }
+        HttpClient httpClient = createHttpClient();
+
+        String connectorName = debeziumConfig.getName();
+        if (connectorExists(httpClient, connectUrl, connectorName)) {
+          return updateConnector(httpClient, connectUrl, connectorName, connectorConfig);
+        } else {
+          return createConnector(httpClient, connectUrl, connectorName, connectorConfig);
+        }
+
+      } catch (Exception e) {
+        log.error("Failed to configure Debezium connector: {}", e.getMessage(), e);
+        return new DefaultUpgradeStepResult(this.id(), DataHubUpgradeState.FAILED);
+      }
+    };
+  }
+
+  /**
+   * Builds connector configuration by merging database connection details, Kafka settings, and
+   * explicit configuration overrides in that priority order.
+   */
+  @com.google.common.annotations.VisibleForTesting
+  Map<String, Object> buildConnectorConfiguration() {
+    Map<String, Object> config = new HashMap<>();
+
+    injectDatabaseConnection(config);
+    injectKafkaConnection(config);
+
+    if (debeziumConfig.getConfig() != null) {
+      config.putAll(debeziumConfig.getConfig());
+    }
+
+    try {
+      log.info(
+          "Built connector configuration with {} properties",
+          opContext.getObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(config));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+
+    return config;
+  }
+
+  /** Extracts database connection from environment variables and EbeanConfiguration. */
+  private void injectDatabaseConnection(Map<String, Object> config) {
+    String dbConnectionUrl = ebeanConfig.getUrl();
+
+    // Parse JDBC URL to extract hostname, port, and database name
+    if (dbConnectionUrl != null && !dbConnectionUrl.isEmpty()) {
+      // Handle MySQL: jdbc:mysql://hostname:port/database
+      // Handle PostgreSQL: jdbc:postgresql://hostname:port/database
+      String urlPattern = "jdbc:(mysql|postgresql)://([^:/]+)(?::(\\d+))?/([^?]+)";
```

Comment:
This seems undesirable, it limits our support to strictly just mysql and postgresql, while those are the only official ones, Ebean should technically work with other backends as well.